src/Compute/Compute/VirtualMachine/Operation/NewAzureVMCommand.cs

```diff
@@ -454,9 +455,19 @@ public async Task<ResourceConfig<VirtualMachine>> CreateConfigAsync()
                 bool enableAcceleratedNetwork = Utils.DoesConfigSupportAcceleratedNetwork(_client,
                     ImageAndOsType, _cmdlet.Size, Location, DefaultLocation);
 
-                var networkInterface = resourceGroup.CreateNetworkInterfaceConfig(
-                    _cmdlet.Name, _cmdlet.EdgeZone, subnet, publicIpAddress, networkSecurityGroup, enableAcceleratedNetwork);
-
+                ResourceConfig<NetworkInterface> networkInterface;
+                if (string.IsNullOrEmpty(publicIpAddress.Name))
+                {
```

Comment:
I could make this its own method, but I'm concerned that passing the large parameters subnet and networkSecurityGroup will offset the benefit of having just this one code block in its own method.